doc/decisions/DR025-UI-delegation-plugin-interface.md

```diff
@@ -0,0 +1,184 @@
+# DR025 UI delegation plugin interface
+
+- **Status:** Proposed
+- **Impact:** High
+- **Driver:** @feltech
+- **Approver:**
+- **Due date:** 2024-08-12
+- **Outcome:**
+
+
+## Background
+
+OpenAssetIO is an abstraction that aims to be minimally opinionated.
+Asset-managed pipelines do not conform to a single structure — each
+facility will vary in its workflow processes and backend systems, often
+significantly. The OpenAssetIO core library provides an API for common
+operations that a host application may wish to perform, with no
+assumptions made about how that data is referenced or structured on the
+backend.
+
+However, some user-facing operations require knowledge of how a
+particular pipeline works so that the appropriate interface can be shown
+to the user.
+
+For example, using a dedicated asset-centric browser UI, rather than
+browsing for files on a file system, could provide a richer and more
+relevant interface for artists, by filtering for approved assets that
+are relevant for their current task, and presenting useful metadata.
+
+A third-party vendor attempting to implement this in a generic way that
+works across all potential pipeline setups is a daunting task, and it is
+very unlikely to warrant expending engineering effort to do optimally
+when it is far from the primary purpose of the software.
+
+The engineers best placed to implement such an interface are the
+maintainers of the asset management system itself. Indeed, this is
+commonplace. However, typically this means separate secondary
```

Comment:
> Indeed, this is commonplace.

For the intended audience, is this a known fact?

Reply:
Honestly not sure :see_no_evil: .  Every manager integration I've seen has at least something, e.g. a menu item.

Reply:
It sounds like a well known fact.

If it's based on just your observation, rather than something you can reference, perhaps it could be rephrased as something like

"Based on experiences with X, Y, Z, this is commonplace."

Reply:
Just dodged this by removing the statement as part of 071bb98e :grin: